File: Data Structures & Algorithms/two-integer-sum/submission-16.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        numToIndexMap = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in numToIndexMap:
                return [numToIndexMap[diff], i]
            
            numToIndexMap.update({nums[i]: i})

File: Data Structures & Algorithms/two-integer-sum-ii/submission-16.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& s, int target) {
        int l = 0, r = s.size()-1;

        while (l <= r) {
            if (s[l] + s[r] > target && l <= r) {
                r--;
            }
            else if (s[l] + s[r] < target && l <= r) {
                l++;
            }
            else {
                return {l+1, r+1};
            }
        }
    }
};
